src/target/riscv/riscv.c

```diff
@@ -2095,6 +2095,8 @@ static int riscv_effective_privilege_mode(struct target *target, int *v_mode, in
 
 static int riscv_mmu(struct target *target, int *enabled)
 {
+	*enabled = 0;
```

Comment:
Some `return`s in this function are preceded by explicit `*enabled = <value>;` whereas others rely on this initialization at the very top of the function.

For consistency, I would recommend to use the same style for all returns.

Reply:
As you can see there were some execution paths that returned ERROR_OK without actually  initializing the value (like on line 21333). 

So, I guess that this is just a  defensive coding (not the real returned value). That being said - it would not hurt to add a comment to clarify the intention.

Reply:
> I repeat my vote to unify the rest of the function: to remove the extra *enabled = 0 assignment

Ah.. I see. I've misinterpreted the original suggestion. Yes, it would be an improvement.

Reply:
Redundant assignments removed. Addressed.